eng/tools/mgmtreport/devops.go

```diff
@@ -0,0 +1,173 @@
+		if err != nil {
+			return 0, err
+		}
+
+		if coverableLines != 0 {
+			break
+		}
+	}
+
+	return *buildId, nil
+}
+
+func getLogID(buildClient build.Client, buildId int) (int, int, error) {
+	var mockTestLogId int
+	var liveTestLogId int
+
+	result, err := buildClient.GetBuildTimeline(context.Background(), build.GetBuildTimelineArgs{
+		Project: &projectName,
+		BuildId: &buildId,
+	})
+	if err != nil {
+		return mockTestLogId, liveTestLogId, err
+	}
+
+	for _, record := range *result.Records {
+		if mockTestLogId != 0 && liveTestLogId != 0 {
+			break
+		} else if *record.State == build.TimelineRecordStateValues.Completed && *record.Result == build.TaskResultValues.Succeeded {
+			if *record.Name == "Mock Test" && mockTestLogId == 0 {
+				mockTestLogId = *record.Log.Id
+			} else if *record.Name == "Run Tests" && liveTestLogId == 0 {
+				liveTestLogId = *record.Log.Id
+			}
+		}
+	}
+
+	return mockTestLogId, liveTestLogId, nil
+}
+
+func getTestResult(buildClient build.Client, buildId, logId int) (int, int, string, error) {
+	logLines, err := buildClient.GetBuildLogLines(context.Background(), build.GetBuildLogLinesArgs{
+		Project: &projectName,
+		BuildId: &buildId,
+		LogId:   &logId,
+	})
+	if err != nil {
+		return 0, 0, "", err
+	}
+
+	logResult := strings.Join(*logLines, "\n")
+	totalTests := regexp.MustCompile("=== RUN.*/").FindAllString(logResult, -1)
+
+	passedTests := regexp.MustCompile("--- PASS:.*/").FindAllString(logResult, -1)
+
+	coverages := regexp.MustCompile("coverage:.*").FindAllString(logResult, -1)
+
+	var coverage string
+	if len(coverages) > 0 {
+		for _, s := range strings.Split(coverages[0], " ") {
+			if strings.Contains(s, "%") {
+				coverage = s
+			}
+		}
+	}
+
+	return len(totalTests), len(passedTests), coverage, nil
+}
+
+func getSkippedMockTest(buildClient build.Client, buildId, mockTestLogId int) (int, error) {
+	logLines, err := buildClient.GetBuildLogLines(context.Background(), build.GetBuildLogLinesArgs{
+		Project: &projectName,
+		BuildId: &buildId,
+		LogId:   &mockTestLogId,
+	})
+	if err != nil {
+		return 0, err
+	}
+
+	logResult := strings.Join(*logLines, "\n")
+	skipOperations := regexp.MustCompile("--- SKIP:.*/").FindAllString(logResult, -1)
+
+	return len(skipOperations), nil
+}
+
+func getCallOperations(buildClient build.Client, buildId, liveTestLogId int) (int, error) {
+	logLines, err := buildClient.GetBuildLogLines(context.Background(), build.GetBuildLogLinesArgs{
+		Project: &projectName,
+		BuildId: &buildId,
+		LogId:   &liveTestLogId,
+	})
+	if err != nil {
+		return 0, err
+	}
+
+	logResult := strings.Join(*logLines, "\n")
+	result := regexp.MustCompile("Call operation:.*").FindAllString(logResult, -1)
+
+	callOperation := make(map[string]int)
+	for _, co := range result {
+		_, after, b := strings.Cut(co, "Call operation: ")
+		if b {
+			if _, ok := callOperation[after]; !ok {
+				callOperation[after] = 0
+			}
+		}
+	}
+
+	return len(callOperation), nil
```

Comment:
This logic is not correct. Live test may call different operations in different RPs. So we need to add two logics: 1. remove duplication, 2. filter the operation does not belong to this RP.